src/chatd.cpp

```diff
@@ -1606,6 +1647,117 @@ void Chat::initChat()
     mHaveAllHistory = false;
 }
 
+void Chat::requestRichLink(Message &message)
+{
+    std::string text = message.toText();
+    std::string url;
+    if (Message::hasUrl(text, url))
+    {
+        std::regex expresion("^(http://|https://)(.+)");
+        std::string linkRequest = url;
+        if (!regex_match(url, expresion))
+        {
+            linkRequest = std::string("http://") + url;
+        }
+
+        auto wptr = weakHandle();
+        Idx messageIdx = msgIndexFromId(message.id());
+        uint16_t updated = message.updated;
+        client().karereClient->api.call(&::mega::MegaApi::requestRichPreview, linkRequest.c_str())
+        .then([wptr, this, messageIdx, updated, text](ReqResult result)
```

Comment:
I think it's safer to capture the `msgid` rather than the index, since a `REJECT` for `JOINRANGEHIST` may clear the history and the captured index may reference a different message.